Backend/routes/routes.py

```diff
@@ -405,54 +412,88 @@ async def delete_supplement_details(id: str):
-    result = ProgramsCollection.replace_one(
-        {"_id": ObjectId(program_id)}, program_dict)
-    if result.matched_count == 0:
-        raise HTTPException(status_code=404, detail="Program not found")
-    program.id = program_id
-    return program
-
-
-@program_router.delete("/programs/{program_id}")
-async def delete_program(program_id: str):
-    result = ProgramsCollection.delete_one({"_id": ObjectId(program_id)})
-    if result.deleted_count == 0:
-        raise HTTPException(status_code=404, detail="Program not found")
-    return {"message": "Program deleted successfully"}
+@program_router.post("/api/add_program")
+async def add_program(program: Program):
+    donor_details = DonorsCollection.find_one({"_id": ObjectId(program.donor_id)})
+    suplement_details = SupplementDetailsCollection.find_one({"_id": ObjectId(program.supplements_details_id)})
+    if not donor_details and suplement_details:
+        raise HTTPException(status_code=404, detail="input are missing")
+    _id = ProgramsCollection.insert_one(dict(program))
+    added_programs = program_list_serializer(ProgramsCollection.find(
+        {"_id": _id.inserted_id}))
+    return {"status": "ok", "data": added_programs}
+
+
+@program_router.get("/api/get_programs")
+async def get_programs():
+    programs = program_list_serializer(
+        ProgramsCollection.find())
+    return {"status": "ok", "data": programs}
+
+
+@program_router.get("/api/get_program/{program_id}")
+async def get_program(id: str):
+    result = program_list_serializer(
+        ProgramsCollection.find({"_id": ObjectId(id)}))
+    return {"status": "ok", "data": result}
+
+
+@program_router.put("/api/update_program/{program_id}")
+async def update_program(id: str, program: Program):
+    ProgramsCollection.find_one_and_update({"_id": ObjectId(id)},
+                                           {"$set": dict(program)})
+    updated_value = program_list_serializer(
+        ProgramsCollection.find({"_id": ObjectId(id)}))
+    return {"status": "ok", "data": updated_value}
+
+
+@program_router.delete("/api/delete_program/{program_id}")
+async def delete_program(id: str):
+    ProgramsCollection.find_one_and_delete({"_id": ObjectId(id)})
+    return {"status": "ok", "data": []}
+
+
+@program_joining.post("/api/Add_program_joining")
+async def add_program_joining(programs: ProgramJoining):
+    result = ProgramsCollection.find_one({"invite_code": programs.invite_code})
+    if not result:
+        raise HTTPException(status_code=404, detail="Invalid invite code")
+    else:
+        programs.program_id = str(result['_id'])
+        _id = ProgramJoiningCollection.insert_one(dict(programs))
+        add_joining_details = programjoining_list_serializer(ProgramJoiningCollection.find({"_id": _id.inserted_id}))
+        return {"status": "ok", "data": add_joining_details}
+
+
+@program_joining.get("/api/get_programs_details")
+async def get_programs():
```

Comment:
the method name says get_programs and url too get_programs..... but in the serializer and collection we are using programjoining so can you rename the method and url matching to it. so that it is readable and easy to understand

Reply:
used for testing only, will be removed